### PR TITLE
Idea 16 support

### PR DESCRIPTION
**Pull request by eliasku**
First attempt...

### CODE REVIEW

**Comment by eliasku** [comment]
For travis checking

**Comment by eliasku** [comment]
Ready....

**Comment by bynuff** [comment]
+++++++

**Comment by EBatTiVo** [comment]
@eliasku - Did you happen to take a look at the v16_BuildSupport branch?  I don't have time to look right at the moment, but I think you may have duplicated some of the work.  Also Issue #454 has a bunch of conversation about what I was doing and where things were left off.

**Comment by eliasku** [comment]
@EBatTiVo sorry, did not check that. I had problems with running plugin on fresh 16 idea install, so I just started fix compatibility for me :)

BTW, is branch-version is ready to go?

**Comment by EBatTiVo** [comment]
Not quite.  Otherwise I would have released it.  Check the issue I gave above for more details.

**Comment by eliasku** [comment]
Checked branch - cool.

My status: plugin is successfully built, runned and tests are passed from IDEA16 runtime. But all Ant tasks are failed, and that's why travis on 1.8 jdk will fail. Tried to change Jre, fix warnings, but nothing helps me. So at least I've solved my personal issue to get working plugin on v16 (packaged by hands).

Class Wrapper: I decided to not use reflection. Just create additional compat source path for version differences. "Compatability" class path will  define the missing class and remap it to new package via inheritance.

I know we all use different versions of IDEA. And we already have nice multi-version packaging / CI. But what the reason to provide new version of plugin with a couple of fixes for old versions of IDEA, which also IDEA don't continue to fix.

I think it would be good for new releases to just fit the latest IDEA version. No problems with multi-JRE version. Easier to test the IDEA target with just one version. Easier and faster CI. No compatibility workarounds and code.

If you need to use haxe for idea-15 - OK, just download the 0.9.10 release. Wants to get new fixes from 0.9.11? ok - just install idea-16. No licence? Use 16 Community edition. If you still more comfortable with idea-15, just use stable idea-15-tested 0.9.10 version.

@EBatTiVo @bynuff @as3boyan What do you think about all of that?

**Comment by EBatTiVo** [comment]
@eliasku - Most developers at TiVo continue to use IDEA 14 versions.  It's not just about licensing cost, but giving developers a more stable platform from which to work.  Always upgrading has its own costs and we prefer to avoid those.  For us, the cost of maintaining multiple versions (of the plugin) is less than the cost of constantly upgrading IDEA.  That's why we have developed the multi-release compatibility layers.

As far as the src/compat directory tree goes... We had that already with v13 and 14.  I'm slowly trying to remove those by using reflection.  If you've got a version 16 running, (hopefully as a set of changes to the v16_BuildSupport branch), then I'll try to merge them.  I may just take them, or I may rework them a bit to get a single source. (I haven't decided yet.)

As far as getting Travis to work, there are a couple of outstanding tasks, but the main one for the v16 build is -- as you found out -- v16 requires Java 8 and everything before it requires java 7.  I haven't looked into that too much, except to say that there are IDEA issues when trying to use a java 8 compiler against v15 and earlier -- even targeting jvm 6 byte code.  I had filed a bug with JetBrains, but I haven't received any new messages since the bug was assigned.  So, we've got to tweak the Travis scripts so that the correct version of java is used for each build.  That is, we need to switch between java 7 and 8 depending upon the version of IDEA we are building against.

At any rate, I know our users are itching for a v16 build and I'd love to give them a release, so thanks for the effort.

File: idea_v16.properties [hunk 1]
```diff
@@ -0,0 +1,47 @@
+#
+# Copyright 2000-2013 JetBrains s.r.o.
+# Copyright 2014-2016 AS3Boyan
+# Copyright 2014-2014 Elias Ku
```
Comment (SlavaRa) [inline idea_v16.properties]: @eliasku maybe 2016-2016?

Comment (eliasku) [inline idea_v16.properties]: No, I've started maintaining plugin with @as3boyan in 2014, then get delayed, and returned by the end of 2015, so 2014, 2016 will be correct :)

Comment (SlavaRa) [inline idea_v16.properties]: but not 2014-2014))

**Comment by alagatar** [comment]
Please share plugin that working with idea 16. Thanks

**Comment by eliasku** [comment]
@alagatar  https://github.com/eliasku/intellij-haxe/releases/tag/v0.10.0-alpha

**Comment by alagatar** [comment]
@eliasku Thanks)

**Comment by eliasku** [comment]
Declined

**Comment by eliasku** [comment]
@alagatar check out https://github.com/eliasku/intellij-haxe/releases/tag/v0.10.0-beta
here 2016.2.3 support update

**Pull request closed without merging**